### PR TITLE
https://tools.hmcts.net/jira/browse/RWA-173

### PR DESCRIPTION
**Pull request by iguannago**
https://tools.hmcts.net/jira/browse/RWA-173

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java [hunk 1]
```diff
@@ -18,7 +18,8 @@
 @RestController
 public class CreateTaskController {
 
-    private final TaskService taskService;
+    @Autowired
+    private TaskService taskService;
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java:22]: we said we were going to do this with constructor autowiring so should be 

public CreateTaskController(TaskService taskService) {
        this.taskService = taskService;
    }

we don't need the the @Autowired that was on there before.

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java [hunk 2]
```diff
@@ -27,21 +28,18 @@
...
     })
-    public ResponseEntity createTask(@RequestBody CreateTaskRequest createTaskRequest) {
+    public ResponseEntity<Object> createTask(@RequestBody CreateTaskRequest createTaskRequest) {
         if (taskService.createTask(
             createTaskRequest.getServiceDetails(),
             createTaskRequest.getTransition(),
             createTaskRequest.getCaseId(),
             createTaskRequest.getDueDate()
         )) {
             return created(null).build();
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java:38]: We should probably return a url to the task management api now with the get task endpoint but that should be another a story and not urgent.

**Review by chrisgrimble — CHANGES_REQUESTED** [review]
The changes for error handling looks good but we did say we would use constructor injection to autowire classes not annotate the fields.

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java (hunk 1 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java]: yes, done.

File: src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java (hunk 2 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/waworkflowapi/controllers/startworkflow/CreateTaskController.java]: yes, it's not defined yet. Let's leave for later on, then.

**Review by chrisgrimble — APPROVED** [review]
LGTM

**Pull request closed without merging**